### PR TITLE
adds serializer for akka.Done. #23854

### PR DESCRIPTION
**Pull request by patriknw**
Finished https://github.com/akka/akka/pull/23855

not enabled by default, but documented

Refs #23854

### CODE REVIEW

File: akka-actor/src/main/scala/akka/Done.scala [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 
+import java.io.Serializable
+
 /**
  * Typically used together with `Future` to signal completion
  * but there is no actual value completed. More clearly signals intent
  * than `Unit` and is available both from Scala and Java (which `Unit` is not).
  */
-sealed abstract class Done
+sealed abstract class Done extends Serializable
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/Done.scala]: `@DoNotInherit`

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-actor/src/main/scala/akka/Done.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor/src/main/scala/akka/Done.scala]: ok, added

**Comment by patriknw** [comment]
ssh issue, I think

**Comment by patriknw** [comment]
PLS BUILD

**Comment by patriknw** [comment]
@renatocaval reminder; you can enable akka.Done serialization in Lagom 1.4 when you start using upcoming Akka 2.5.8

**Comment by renatocaval** [comment]
@patriknw thanks, I will take note
/cc @TimMoore  @ignasi35

**Comment by patriknw** [comment]
however, if you want to support rolling upgrades between 1.3 and 1.4 you can't enable it by default, because then during the upgrade a new 1.4 node can send Done with the new serializer to an 1.3 node that doesn't understand that

**Comment by TimMoore** [comment]
Thanks @patriknw and @renatocaval.

Would you consider a backport to Akka 2.4.x similar to what we discussed in https://github.com/akka/akka/issues/24009#issuecomment-347938263? While I argued against a backport in that issue (which was about changing shard coordinator mode), this seems like a smaller and more straightforward change.

We've added a few other serialization changes to Lagom already, so we'll need to be sure we use a consistent approach. With other changes, we have backported serializers to allow for a seamless upgrade path from the latest 1.3.x to 1.4.0. An alternative would be to extract this serializer from Akka and add it directly to Lagom 1.3.x (as I suggested in https://github.com/lagom/lagom/pull/1068#issuecomment-340974577) but it might be more useful to have it available in Akka 2.4.x directly if that's a possibility. I raised https://github.com/lagom/lagom/issues/1133 to document this change in Lagom's migration guide, etc. We can continue discussion there.

**Comment by renatocaval** [comment]
@patriknw any chance to backport this to Akka 2.4.x?

**Pull request merged**